ccp/src/utility_thread/thread.rs

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use futures::FutureExt;
+use tokio::sync::mpsc;
+use tokio::sync::oneshot;
+
+use crate::utility_thread::proof_storage::ProofStorage;
+use crate::utility_thread::UTError;
+use ccp_shared::proof::CCProof;
+use ccp_shared::proof::CCProofId;
+use ccp_shared::proof::ProofIdx;
+use ccp_shared::types::GlobalNonce;
+use cpu_utils::LogicalCoreId;
+
+use super::message::*;
+use super::UTResult;
+
+type ThreadShutdownInlet = oneshot::Sender<()>;
+type ThreadShutdownOutlet = oneshot::Receiver<()>;
+
+pub(crate) struct UtilityThread {
+    to_utility: ToUtilityInlet,
```

Comment:
I think the name should be say about the source of these messages (i.e. prover output), as the messages are quite concrete.

Reply:
The idea was, that every component can send a message into the utility thread, that why the name shows only direction.